apps/tx-builder/src/components/utils/Confirmable.tsx

```diff
@@ -0,0 +1,78 @@
+import ReactDOM from 'react-dom'
+import React from 'react'
+import { ThemeProvider } from 'styled-components'
+import { theme } from '@gnosis.pm/safe-react-components'
+
+type ConfirmationProps = {
+  proceed: (value: boolean) => void
+  show: boolean
+}
+
+type Confirmable = {
+  resolve: (value: boolean) => void
+}
+
+const confirmable =
```

Comment:
Do you think that this component is needed? we have modals like `DeleteBatchModal`, `SaveBatchModal` that works without this logic.

Reply:
Yes, they work but in much simpler flows. I wanted to have something similar to js' `confirm` built-in